components/AnimatedImage/library/src/desktopMain/kotlin/org/jetbrains/compose/animatedimage/NetworkLoader.kt

```diff
@@ -0,0 +1,27 @@
+package org.jetbrains.compose.animatedimage
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.net.HttpURLConnection
+import java.net.URL
+
+internal class NetworkLoader(private val imageUrl: String) : AnimatedImageLoader {
+    var cachedBytes: ByteArray? = null
+
+    override suspend fun loadBytes(): ByteArray = withContext(Dispatchers.IO) {
+        var bytesArray: ByteArray? = cachedBytes
+
+        if (bytesArray == null) {
+            bytesArray = loadNetworkResource(imageUrl)
```

Comment:
Maybe we simplify it into `URL(imageUrl).readBytes()`? We don't need another withContext here.